File: packages/sdui-common/src/main/java/nl/q42/common/components/TextFormatting.java
package nl.q42.common.components;

public enum TextFormatting
{
  NORMAL,
  BOLD,
  ITALIC,
  UNDERLINE;

  public TextFormatting from(int value)
  {
    var values      = TextFormatting.values();
    var withinRange = value >= 0 && value < values.length;

    return !withinRange
           ? NORMAL : values[value];
  }
}

File: packages/sdui-common/src/commonMain/kotlin/nl/q42/common/components/TextFormatting.kt
package nl.q42.common.components

enum class TextFormatting {
    NORMAL,
    BOLD,
    ITALIC,
    UNDERLINE;

    fun from(value: Int): TextFormatting? {
        val values: Array<TextFormatting?> = TextFormatting.entries.toTypedArray()
        val withinRange = value >= 0 && value < values.size

        return if (!withinRange) TextFormatting.NORMAL else
            values[value]
    }
}
